### PR TITLE
feat: Repo entity now reports last push and last main branch push dates

### PR DESCRIPTION
**Pull request by jpmcb**
## Description

The repo entity (like when doing `v2/repos/search` will return the last push and last main branch push dates.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes: https://github.com/open-sauced/api/issues/657

## Steps to QA

When using `v2/repos/search` for `open-sauced/app`:

```json5
{
  "data": [
    {
      "id": 501028599,

      // etc. etc.

      "last_pushed_at": "2024-03-27T19:49:01.000Z",
      "last_main_pushed_at": "2024-03-27T17:09:21.000Z"
    }
  ],
  "meta": {
    "page": 1,
    "limit": 10,
    "itemCount": 1,
    "pageCount": 1,
    "hasPreviousPage": false,
    "hasNextPage": false
  }
}
```

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: src/timescale/push_github_events.service.ts [hunk 1]
```diff
@@ -74,4 +75,38 @@ export class PushGithubEventsService {
...
+      return { push_date: new Date(0), main_push_date: new Date(0) };
+    }
+
+    const firstPushDate = results[0].event_time;
+    const mainMasterRegex = new RegExp("main|master");
+    let firstMainPushDate = results.find((item) => item.push_ref && mainMasterRegex.test(item.push_ref))?.event_time;
+
+    if (!firstMainPushDate) {
+      firstMainPushDate = new Date(0);
+    }
```
Comment (nickytonline) [inline src/timescale/push_github_events.service.ts:105]: refactor (non-blocking): Consider using nullish coalescing instead, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing.

```suggestion
    const firstMainPushDate = results.find((item) => item.push_ref && mainMasterRegex.test(item.push_ref))?.event_time;
 ?? new Date(0);
```

**Review by nickytonline — APPROVED** [review]
Looks great John! :shipit:

File: src/timescale/push_github_events.service.ts (hunk 1 above)
Comment (jpmcb) [reply src/timescale/push_github_events.service.ts:105]: 💯

**Comment by jpmcb** [comment]
Force pushed to address Nick's comment and fix test dependency injection

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Ready to go, just needs merge conflict resolution.

**Pull request merged**